web/src/selectors/Assertion.selectors.ts

```diff
@@ -17,9 +17,46 @@ const selectAffectedSpanList = createSelector(stateSelector, paramsSelector, (st
   return trace?.spans.filter(({id}) => spanIdList.includes(id)) || [];
 });
 
+const selectResultAssertionsBySpan = createSelector(
+  stateSelector,
+  (_: RootState, testId: string, runId: string, spanId: string) => ({testId, runId, spanId}),
+  (state, {testId, runId, spanId}) => {
+    const {data} = endpoints.getRunById.select({testId, runId})(state);
+
+    if (!data) return {};
+
+    // Map and flat items in one single array
+    const results = data.result.resultList
+      .flatMap(selector =>
+        selector.resultList.map(assertion => ({
+          id: selector.selector,
+          attribute: assertion.assertion.attribute,
+          label: `${selector.selectorList.map(({value}) => value).join(' ')} ${
+            selector.pseudoSelector?.selector ?? ''
+          }`,
+          result: assertion.spanResults.find(spanResult => spanResult.spanId === spanId),
+        }))
+      )
+      // Filter if it has result for the spanId
+      .filter(assertion => Boolean(assertion?.result))
+      // Hash items by attribute
+      .reduce((prev: {[key: string]: any}, curr) => {
```

Comment:
I tend to avoid using any, is this necessary or can we type everything that is part of this flow?

Reply:
Done